src/compute/src/rpc/service/stream_service.rs

```diff
@@ -192,39 +188,9 @@ impl StreamService for StreamServiceImpl {
                 |err| tracing::error!(error = %err.as_report(), "failed to collect barrier"),
             )?;
 
-        let (synced_sstables, table_watermarks) = match kind {
-            BarrierKind::Unspecified => unreachable!(),
-            BarrierKind::Initial => {
-                if let Some(hummock) = self.env.state_store().as_hummock() {
-                    let mce = hummock.get_pinned_version().max_committed_epoch();
-                    assert_eq!(
-                        mce, req.prev_epoch,
-                        "first epoch should match with the current version",
-                    );
```

Comment:
we dont need the check anymore?

Reply:
The code is moved to `managed_state::sync_epoch`. The check is still there.